### PR TITLE
fixed type filter accepting tiny and small dusts with the dust filter

### PR DESCRIPTION
**Pull request by Trinsdar**

### CODE REVIEW

**Comment by e99999** [comment]
have bear test it first, this is exactly what I tried when i fixed it. if it doesnt work you need to make the filter string a network feild mabye? Ill merge the PR when bear confirms it works properly

**Comment by Trinsdar** [comment]
Ok

**Comment by e99999** [comment]
do crushed /crushPurified and ill merge?

File: src/main/java/gtclassic/common/util/GTIFilters.java [hunk 1]
```diff
@@ -123,8 +123,15 @@ public boolean matches(ItemStack stack) {
...
-					: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());
+			if (this.tile.getCurrentFilter().equals("dust")){
+				return this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) || GTHelperStack.oreDictStartsWith(stack, "dustTiny") || GTHelperStack.oreDictStartsWith(stack, "dustSmall")
+						: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) && !GTHelperStack.oreDictStartsWith(stack, "dustTiny") && !GTHelperStack.oreDictStartsWith(stack, "dustSmall");
+			}
+			if (this.tile.getCurrentFilter().equals("ingot")){
+				return this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) || GTHelperStack.oreDictStartsWith(stack, "ingotHot")
+						: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) && !GTHelperStack.oreDictStartsWith(stack, "ingotHot");
+			}
+			return this.tile.invertFilter != GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());
```
Comment (e99999) [inline src/main/java/gtclassic/common/util/GTIFilters.java:134]: what is this last line? this makes no sense

Comment (Trinsdar) [reply src/main/java/gtclassic/common/util/GTIFilters.java:134]: oh that was a sonar lint suggestion. if you think about it, it is basically the same as 
```
this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter())
					: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());``` 
lets use dust for an example
if invert filter is false, and the stack is not dust(false), it returns false.
if invert filter is false and the stack is dust(true), it returns true.
if invert filter is true, and the stack is not dust(false), it returns true.
if invert filter is true, and the stack is dust(true), it returns false
so therefore that logic is actually the same and cleaner

**Pull request merged**